pkg/controllers/updaterun/controller.go

```diff
@@ -268,17 +268,22 @@ func (r *Reconciler) SetupWithManager(mgr runtime.Manager) error {
 		Named("clusterresource-stagedupdaterun-controller").
 		For(&placementv1beta1.ClusterStagedUpdateRun{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Watches(&placementv1beta1.ClusterApprovalRequest{}, &handler.Funcs{
-			// We only care about when an approval request is approved.
+			// We watch for ClusterApprovalRequest to be approved.
 			UpdateFunc: func(ctx context.Context, e event.UpdateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
 				klog.V(2).InfoS("Handling a clusterApprovalRequest update event", "clusterApprovalRequest", klog.KObj(e.ObjectNew))
-				handleClusterApprovalRequest(e.ObjectOld, e.ObjectNew, q)
+				handleClusterApprovalRequestUpdate(e.ObjectOld, e.ObjectNew, q)
+			},
+			// We watch for ClusterApprovalRequest deletion events to recreate it ASAP.
```

Comment:
why do u we care if the approval is already approved? do we not care if it's already approved or not?